app/src/main/java/com/example/sharingang/DetailedItemFragment.kt

```diff
@@ -73,6 +79,62 @@ class DetailedItemFragment : Fragment() {
         return binding.root
     }
 
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+        super.onCreateOptionsMenu(menu, inflater)
+        inflater.inflate(R.menu.menu_detailed, menu)
+    }
+
+    override fun onPrepareOptionsMenu(menu: Menu) {
+        super.onPrepareOptionsMenu(menu)
+        val sell = menu.findItem(R.id.menuSell)
+        val resell = menu.findItem(R.id.menuResell)
+        if (!args.item.userId.equals(currentUserProvider.getCurrentUserId())) {
+            menu.findItem(R.id.menuEdit).isVisible = false
+            sell.isVisible = false
+            resell.isVisible = false
+        } else {
+            sell.isVisible = !(item?.sold ?: args.item.sold)
+            resell.isVisible = item?.sold ?: args.item.sold
+        }
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        return when (item.itemId) {
+            R.id.menuEdit -> {
+                view?.findNavController()?.navigate(
+                    DetailedItemFragmentDirections.actionDetailedItemFragmentToNewEditFragment(args.item)
+                )
+                true
+            }
+            R.id.menuSell -> {
+                updateSold()
+                true
+            }
+            R.id.menuResell -> {
+                updateSold()
+                true
+            }
+            else -> super.onOptionsItemSelected(item)
+        }
+    }
+
+    private fun updateSold() {
+        lifecycleScope.launch(Dispatchers.IO) {
+            itemViewModel.sellItem(item)
+            getItem()
+        }
+    }
+
+    private fun getItem() {
+        lifecycleScope.launch(Dispatchers.IO) {
+            item = itemRepository.get(args.item.id!!)
+            lifecycleScope.launch(Dispatchers.Main) {
+                binding.item = item
+                activity?.invalidateOptionsMenu()
+            }
+        }
+    }
```

Comment:
Is this ever used outside of a coroutine scope (like it is in `updateSold`). If so, consider just making this `private suspend fun getItem()`, and then just calling it directly inside of `updateSold`, to avoid needlessly launching an additional scope

Reply:
Yeah that's a good point, I'll see if it works that way because it's indeed better.

Reply:
It is called directly in onCreateView, so outside a coroutine scope.

Reply:
F